src/circuit/blinding_circuit.rs

```diff
@@ -34,69 +37,89 @@ where
         // parse the public inputs (todo is Com(Z_H) a public input?)
         let com_z_h = TEGroupAffine::<CP::Curve>::new(self.zh[0], self.zh[1]);
 
-        assert_eq!(self.private_inputs.len() % 3, 0);
-        let mut i = 0;
-        let mut q: Point<CP::Curve>;
-        while i < self.private_inputs.len() {
-            // parse the private inputs
-            q = composer.add_affine(TEGroupAffine::<CP::Curve>::new(
-                self.private_inputs[i],
-                self.private_inputs[i + 1],
-            ));
-            let b = composer.add_input(self.private_inputs[i + 2]);
-            // constraints
+        let blind_vec = vec![
+            self.blinding.q_m,
+            self.blinding.q_l,
+            self.blinding.q_r,
+            self.blinding.q_o,
+            self.blinding.q_4,
+            self.blinding.q_c,
+        ];
+        assert_eq!(blind_vec.len(), BLINDING_PC_NUM);
+        let vp_desc = self.vp_desc.get_pack().unwrap();
+
+        // Constrain vp blinding
+        for (point, blind) in vp_desc[0..2 * BLINDING_PC_NUM]
```

Comment:
The `blind` is not in the "description of the VP", it is from `.zip(blind_vec.iter())`.
`blind_vec` is for the blinding scalar for q_L, q_R, etc.
We don't have commitment for `vp_desc` yet. It might be `Com(Com_q(vp_desc), r)`, in which we need one random scalar r.